tasks/request_status_for_granule/request_status_for_granule.py

```diff
@@ -55,21 +63,11 @@ def task(
     if job_id is None or len(job_id) == 0:
         job_id = get_most_recent_job_id_for_granule(granule_id, engine)
         if job_id is None:
-            return create_http_error_dict(
-                "NotFound",
-                HTTPStatus.NOT_FOUND,
-                request_id,
-                f"No job for granule id '{granule_id}'.",
-            )
+            raise JobNotFoundException(granule_id)
 
     job_entry = get_job_entry_for_granule(granule_id, job_id, engine)
     if job_entry is None:
-        return create_http_error_dict(
-            "NotFound",
-            HTTPStatus.NOT_FOUND,
-            request_id,
-            f"No job found for granule id '{granule_id}' and job id '{job_id}'.",
-        )
+        raise JobGranuleCombinationNotFoundException(job_id, granule_id)
```

Comment:
Pulled AWS specific error-handling out to handler function here and in request_status_for_job.
As-is, handler didn't know if it should check against the output.json or if an error occurred.